documentation/modules/ROOT/pages/architecture.adoc

```diff
@@ -30,13 +30,13 @@ As shown in the image, the {prodname} connectors for MySQL and PostgresSQL are d
 Kafka Connect operates as a separate service besides the Kafka broker.
 
 By default, changes from one database table are written to a Kafka topic whose name corresponds to the table name.
-If needed, you can adjust the destination topic name by configuring {prodname}'s {link-prefix}:{link-topic-routing}[topic routing transformation]. For example, you can: 
+If needed, you can adjust the destination topic name by configuring {prodname}'s xref:{link-topic-routing}#topic-routing[topic routing transformation]. For example, you can:
 
 * Route records to a topic whose name is different from the table's name
 * Stream change event records for multiple tables into a single topic
 
 After change event records are in Apache Kafka, different connectors in the Kafka Connect eco-system can stream the records to other systems and databases such as Elasticsearch, data warehouses and analytics systems, or caches such as Infinispan.
-Depending on the chosen sink connector, you might need to configure {prodname}'s {link-prefix}:{link-event-flattening}[new record state extraction] transformation. This Kafka Connect SMT propagates the `after` structure from {prodname}'s change event to the sink connector. This is in place of the verbose change event record that is propagated by default. 
+Depending on the chosen sink connector, you might need to configure {prodname}'s xref:{link-event-flattening}#new-record-state-extraction[new record state extraction] transformation. This Kafka Connect SMT propagates the `after` structure from {prodname}'s change event to the sink connector. This is in place of the verbose change event record that is propagated by default.
```

Comment:
```suggestion
Depending on the chosen sink connector, you might need to configure {prodname}'s xref:transformations/event-flattening.adoc#new-record-state-extraction[new record state extraction] transformation. This Kafka Connect SMT propagates the `after` structure from {prodname}'s change event to the sink connector. This is in place of the verbose change event record that is propagated by default.
```

Why not just this? I.e. I was hoping we'd get rid of these link variables altogether? I see how that'll mean a bit more work when moving a page from one section to another, but then I don't think we'd do this very often. So I'd prefer to keep things simple and just have the plain xref link, no variables whatsoever. WDYT, @roldanbob?

Reply:
Oh, I begin to understand now. How is this then working elsewhere though? If for instance you look at the second link in [this section](https://github.com/debezium/debezium/blob/main/documentation/modules/ROOT/pages/transformations/content-based-routing.adoc#options-for-applying-the-transformation-selectively):

    xref:transformations/filtering.adoc#filter-topic-regex[topic.regex]

Is this actually failing to render downstream then?

Reply:
Ok, I had to do some investigation, because I wasn't sure what was going on, because in the downstream doc, the second link that you point to is actually working, which didn't make sense to me. 

But if you look at [that line in the 1.6 and 1.7 branches](https://github.com/debezium/debezium/blame/1.7/documentation/modules/ROOT/pages/transformations/content-based-routing.adoc#L171), you can see that they both use the format `xref:filter-topic-regex[topic.regex]`. So somehow, the change was not propagated to the `main` branch. We first introduced the SMT predicates in 1.5 (downstream 2021.q3), and the link works in that version. I wonder whether things got out of synch with the change from using the `master` branch?

In any case, for that second link (and similar links in other SMT topics) the upstream source _should_ use a simple xref, but it needs to be based on the **anchor ID** only, and should not include the file path.

Reply:
I just checked the other two files where there is a similar link, `flitering.adoc` and `topic-routing.adoc`, and they both already use the correct `xref:_<anchorID>_[_<link-text>_]` format.

Reply:
Ok, thanks.